### PR TITLE
Code review: Ruby on Rails Forms

### PR DESCRIPTION
**Pull request by fdfragoso**
All the requirements from Microverse and The Odin Project were implemented, including the extra credit (show full error messages).

Linter and Rubocop added

Waiting for code review

### CODE REVIEW

**Review by BrittanyBlake — COMMENTED** [review]
### Status: Invalid 🛑 
****
Hi @fdfragoso  and @keddo. This pull request has been marked as invalid because you are missing a [descriptive README](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243642/). Please ensure that you have a descriptive README file before you submit your next pull request. 

- Additionally, you are missing a master branch. Please note that proper GitFlow means that you should be merging from feature --> development --> master. 

****
Please make the required adjustments and submit a new code review request.

### Good Luck and Happy Coding 🍀 👩🏾‍💻

File: app/views/users/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      <li><%= error %></li>
+    <% end %>
+</ul>
+
+<%= form_for(@user) do |f| %>
+  User name: <%= f.text_field :username, placeholder: "Your user name here" %><br />
+  Email: <%= f.email_field :email, placeholder: "example@example.com"%><br />
+  Password: <%= f.password_field :password %><br />
+  <%= f.submit "" %>
+<% end %>
```
Comment (trekab) [inline app/views/users/new.html.erb]: Using the empty quotes(`""`) in the submit button leaves your submit button without a label.

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
+class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def create
+    # @user = User.new(username: params[:username], email: params[:email], password: [:password])
```
Comment (trekab) [inline app/controllers/users_controller.rb]: Remove this commented code. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  def edit
+    @user = User.find(params[:id])
+  end
+
+  def update
+    @user = User.find(params[:id])
+    @user.update(user_params)
+
+    # flash.notice = "Article '#{@article.title}' Updated!"
+
```
Comment (trekab) [inline app/controllers/users_controller.rb]: Remove this commented code. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))

**Review by trekab — CHANGES_REQUESTED** [review]
### Changes Required! :recycle: 
Hi @fdfragoso and @keddo! Nice job so far. You have a few changes to make to have this project completed. These are:
- [x] For the `UsersController`, the `update` action should either `redirect_to new_user_path` if the user was saved ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_create.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms)) or `render :new` if the user was not saved ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_create.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))
- [x] You were expected to leave the commented out HTML form for this review. This is to justify that you followed the learning steps outlined in the Odin project task. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_form2.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))
- [x] __[Optional]__ Create a partial for the user form and reuse it in both the edit and new view. This would be in line with the [DRY principle](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself).
- [ ] Check out the comments in the code for more feedback.

__Kindly make the necessary changes and submit another code review request.__
_Feel free to reach me on slack for any inquiries about this review._ :wink: 
__Treasure Kabareebe__
[GitHub](https://github.com/trekab) | [LinkedIn](https://www.linkedin.com/in/treasure-kabareebe/) | [Twitter](https://twitter.com/TKabareebe)

**Review by sumancrest0001 — APPROVED** [review]
**Status: Approved :+1: :1st_place_medal: 
Congratulations TEAM :clap:** 
It is approved as you implemented all the changes requested by the previous TSE. It is working as expected. Now you can move to another project. All the Best.
Happy coding!!!

**Pull request merged**